### PR TITLE
ci: add .github/workflows/conventional-commits.yml

### PR DESCRIPTION
**Pull request by gobengo**
Motivation:
* robot-enforcement of [policy](https://filecoinproject.slack.com/archives/C02BZPRS9HP/p1648067416521459?thread_ts=1648065445.982479&cid=C02BZPRS9HP)

Changes
* Add a github action to enforce that PR titles follow conventional commits (so that squash merges of PRs dont lead to commit messages violating [this policy](https://github.com/nftstorage/nft.storage#how-should-i-write-my-commits))

### CODE REVIEW

File: .github/workflows/conventional-commits.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: 'Check Conventional Commits'
+
+on:
+  pull_request_target:
```
Comment (gobengo) [inline .github/workflows/conventional-commits.yml:4]: I think this is the best [trigger](https://github.com/amannn/action-semantic-pull-request#event-triggers) to use.

A quirk is that this check will only start being enforced on new PRs once this PR is merged into main branch

**Comment by gobengo** [comment]
FWIW I tested this same yaml file in another repo and it successfully blocked a PR without a conventional commit prefix

**Comment by alanshaw** [comment]
@gobengo this sounds good to me. What happens when someone doesn't follow the convention? They just get a failure in checks right? Do they get adequate feedback so they can add the appropriate prefix? I'm just thinking about a first time contributor that isn't familiar with conventional commit messages.

If you have a link to the PR you tested with that would be a helpful preview! 😊

It looks like it from the config, but I assume that when the user edits the title for the PR on github the check is re-run?

**Comment by hugomrdias** [comment]
to make sure i understand, this validates the PR title but does NOT validate PR commits or squash/merge?

**Comment by gobengo** [comment]
@alanshaw 

> What happens when someone doesn't follow the convention? They just get a failure in checks right? Do they get adequate feedback so they can add the appropriate prefix? I'm just thinking about a first time contributor that isn't familiar with conventional commit messages.
> 
> If you have a link to the PR you tested with that would be a helpful preview! 😊

Yep, here is an example failed check when the [PR](https://github.com/gobengo/conventional-commits-repo/pull/1) title does not match conventionalcommits.org syntax: https://github.com/gobengo/conventional-commits-repo/runs/5742325607?check_suite_focus=true

> It looks like it from the config, but I assume that when the user edits the title for the PR on github the check is re-run?

Yep, revisiting the same example:
* I then [changed the PR title by prepending 'docs: '](https://github.com/gobengo/conventional-commits-repo/pull/1#event-6328891122)
* That automatically [triggered the gh action to check again, this time succeeding](https://github.com/gobengo/conventional-commits-repo/runs/5742359589?check_suite_focus=true)

**Comment by gobengo** [comment]
@hugomrdias 

> to make sure i understand, this validates the PR title but does NOT validate PR commits or squash/merge?

Correct.
* The individual commits in [this sample PR](https://github.com/gobengo/conventional-commits-repo/pull/1/commits) do not use conventionalcommits.org formatting, and the check did not fail.
* Left to convention without enforcement for most commits to be created by merging PRs as squash, batching a bunch of not-necessarily-conventional commits messages into one big conventionalcommit for the PR as a whole

This reminded me to test the following intended invariant:
* Commits pushed to 'main' branch that dont use conventionalcommits.org syntax should NOT fail gh action checks
  * This may or may not be desirable to add later, but it's more aggressive than I'd recommend starting out with, and is (I believe) consistent with the status quo
  * Test
    * tested by merging [this pr](https://github.com/gobengo/conventional-commits-repo/pull/1), which added a non-conventional commit into main ([becf9](https://github.com/gobengo/conventional-commits-repo/commit/becf91d7572456a1f41aeb5fcbf6646d9ca019b1))
    * expected result: [no failed actions running against main branch post-merge](https://github.com/gobengo/conventional-commits-repo/actions)

**Comment by gobengo** [comment]
@alanshaw @hugomrdias Any remaining concerns?

**Review by alanshaw — APPROVED** [review]

**Review by hugomrdias — APPROVED** [review]

**Pull request merged**